### ISSUE TITLE
Dumping jobs from storage ignoring priority settings.

### ISSUE THREAD

**Issue by alexperovich** [MaFi-Games/Captain-of-Industry-issues#1212]
### Issue description

It appears that the issue here https://github.com/MaFi-Games/Captain-of-Industry-issues/issues/743 has come back. It was supposed to be fixed in 0.4.13, but I am still seeing it in 0.5.3. I have a storage filling with rock from mining, and have it set as P10 keep empty, but all my trucks are swarming to dump the rock from that storage rather than fill my higher priority storages that feed my smelters.

### Reproduction steps

1. Load save.
2. Observe the trucks prioritizing dumping rock from the storage with priority 10 over filling other storages at priority 7.

### Game version and build number

v0.5.3e b225

### Operating system

Windows

### Attachments

[issue.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/12142419/issue.zip)

**Comment by alexperovich** [comment]
The same thing is happening with construction jobs. A P8 construction job is taking priority over p7+ deliveries.

**Comment by alexperovich** [comment]
It appears that the actual priority numbers don't matter, dumping and construction jobs are always taking priority over deliveries.

**Comment by JamieRCHI** [comment]
> It appears that the actual priority numbers don't matter, dumping and construction jobs are always taking priority over deliveries.

Yep. I think you are right. Would explain my issues when playing the game in the last few days. Trucks seem to be much more 'Busy' than I remember as well.
